sunpy/map/header_helper.py

```diff
@@ -197,14 +209,16 @@ def _get_observer_meta(coordinate):
             * hgln_obs, hglt_obs
             * dsun_obs
             * rsun_obs
+            * rsun_ref
     """
+    observer = observer.transform_to(frames.HeliographicStonyhurst(obstime=observer.obstime))
```

Comment:
is this just to make life easier when pulling out meta?

Reply:
This is to ensure that we are always in HGS before we build the header (which when coordinate was passed in was kind of enforced by the frame).